data-raw/admiral_verify_templates.R

```diff
@@ -0,0 +1,354 @@
+#'   ("inst/directory)
+#' @param cache_dir (cache)  Path to cache.       (varies by OS/configuration)
+#'   ~/.config/R/admiral_templates_data
+#' @param adam_old_dir  temporary directory where old ADaMs
+#'   (downloaded from github.com) are stored
+#' @param adam_new_dir  temporary directory where new ADaMs
+#'   (created by templates) are stored
+
+
+#' @title verify_templates
+#' @param pkg package (currently only admiral)
+#' @param ds  character vector of ADaM names.  Corresponding to templates to run.  \
+#'   A subset of adam_names.
+#' @description:
+#' Generates ADaM from templates and compares to previously generated ADaM file.
+#' (The latter are found in https:://github.com main branch pharmaverseadam).
+#'
+#' Much code taken from pharamavreseadam::create_adams_data.R
+#'  (https://github.com/pharmaverse/pharmaverseadam/blob/main/data-raw/create_adams_data.R)
+#'
+#' USAGE:   verify_templates()
+#'
+#' @export
+verify_templates <- function(pkg = "admiral", ds = c("adae")) {
+  # TODO: delete all remove prior ADaM downloads
+  # ASSUME:  (1) user is running script for 1st time and no temporary directories exist, OR
+  #          (2) user is running script a 2nd time, in same session, and must remove directories
+
+  clean_cache() # clear all..
+
+  pkg <- "admiral"
+  if (pkg != "admiral") cli_abort("Currently only `admiral` package is supported")
+
+  # nolint start
+  library(pkg, character.only = TRUE)
+  library(teal.data)
+  library(purrr)
+  library(cli)
+  # nolint end
+  cli_alert("Generating ADaMs for { pkg} package.")
+
+  # TODO:  do not show Warnings
+  # creae temporary directories
+  cli_inform("Creating temporary directories")
+  x <- tempdir()
+  dir.create(file.path(x, "old"), showWarnings = TRUE)
+  dir.create(file.path(x, "new"), showWarnings = TRUE)
+  dir.create(file.path(x, "diff"), showWarnings = TRUE)
+
+  # TODO:
+  path <- list(
+    template_dir = "inst/templates",
+    cache_dir = tools::R_user_dir("admiral_templates_data", which = "cache"),
+    adam_new_dir = file.path(x, "new"),
+    adam_old_dir = file.path(x, "old"),
+    diff = file.path(x, "diff")
+  )
+
+  # TODO:
+  # if dir exists then empty it ; if not exist then create it.
+  # lapply(path, function(x)  if( x %in% c("template_dir", "cache_dir")!exists(x)) dir.create(x))
+
+  # gather all templates for this pkg (12 found) ----
+  templates <- list.files(path$template_dir, pattern = "ad_")
+
+  # from templates  generate vector of adam_names
+  adam_names <- vapply(templates, function(x) gsub("ad_|\\.R", "", x),
+    USE.NAMES = FALSE, character(length = 1)
+  )
+
+  # check
+  if (length(templates) != length(adam_names)) stop("Number of templates and adam_names differ")
+
+  # templates is a named chr[]
+  names(templates) <- adam_names
+
+  # User specified templates in `ds`. Limit to only these templates.
+
+  adam_names <- adam_names[adam_names %in% ds]
+
+  # per Ben, ignore "adlbhy"
```

Comment:
Is this still relevant? If so:
```suggestion
  # ignore "adlbhy" because XYZ
```
@bms63 what was the reason for this?